boards/PDM/Inc/Io/Io_Efuse_22XS4200.h

```diff
@@ -0,0 +1,646 @@
+ */
+void Io_Efuse_22XS4200_Init(SPI_HandleTypeDef *const hspi);
+
+/**
+ * Configures the Efuse's registers using the values defined in: REGNAME_CONFIG
+ * @param *e_fuse Pointer to E_Fuse structure being configured
+ */
+void Io_Efuse_22XS4200_configureEfuse(struct E_Fuse *e_fuse);
+
+/**
+ * Sets current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  CSNS_FUNCTION_CURRENT_CH0 - Current sensing for channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Ffuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed channels (for Parallel mode)
+ * @param *e_fuse Pointer to E_Fuse structure being configured
+ */
+void Io_Efuse_22XS4200_configure_channel_monitoring(uint8_t selection, struct E_Fuse *e_fuse);
+
+/**
+ * Gets the value of the Efuse's Status Register over SPI and updates the Efuse handle's operating status
+ * @param *e_fuse Pointer to E_Fuse structure being checked
+ */
+void Io_Efuse_22XS4200_UpdateStatus(struct E_Fuse *e_fuse);
+
+/**
+ * DGets the value of the Efuse's Fault Register over SPI and updates the Efuse handle's fault status
+ * @param *e_fuse Pointer to E_Fuse structure being checked
+ */
+void Io_Efuse_22XS4200_UpdateFaults(struct E_Fuse *e_fuse);
+
+
+/**
+ * Write data to a specific Serial Input register on the Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @param *e_fuse Pointer to E_Fuse structure being written to
+ */
+void Io_Efuse_22XS4200_WriteReg(uint8_t register_address, uint16_t register_value, struct E_Fuse *e_fuse);
+
+/**
+ * Read data from a specific Serial Output register on the Efuse
+ * @param register_address Serial Output register being read from
+ * @param *e_fuse Pointer to E_Fuse structure being read from
+ * @return Contents of the addressed register
+ */
+uint16_t Io_Efuse_22XS4200_ReadReg(uint8_t register_address, struct E_Fuse *e_fuse);
+
+/**
+ * Calculates the parity of the SPI command to be sent to the Efuse.
+ * Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit
+ * @return Returns 1 if the parity bit needs to be set, else returns 0
+ */
+uint8_t Io_Efuse_Cal_Parity_Bit(uint16_t spi_command);
+
+/**
+ * Write SPI data to Efuse
+ * @param *TxData Pointer to data being sent to the Efuse
+ * @param *GPIO_Port Handle to Efuse's chip-select GPIO port
+ * @param GPIO_Pin Efuse's chip-select GPIO pin
+ * @return Status of the SPI transfer
+ */
+HAL_StatusTypeDef Io_Efuse_22XS4200_WriteToEfuse(uint16_t *TxData, GPIO_TypeDef *GPIO_Port, uint16_t GPIO_Pin);
```

Comment:
It seems like `Io_Efuse_22XS4200_WriteToEfuse` and `Io_Efuse_22XS4200_ReadFromEfuse` are blocking at the moment? Can we switch to interrupt SPI so it's not blocking? Or at least make a note that it IS blocking at the moment and we can change it to non-blocking later.

Reply:
Ya I've added a comment for now that it's blocking. If we switched to interrupt or DMA based I feel like the call flow would be more confusing, because you'd have to deal with the data in the callback function then instead?

Reply:
I'm fine with blocking for now